niceman/distributions/base.py

```diff
@@ -50,6 +101,38 @@ def yaml_representer(dumper, data):
         return dumper.represent_mapping('tag:yaml.org,2002:map', ordered_items)
 
 
+    def satisfied_by(self, other):
+        """Determine if the other spec object satisfies the requirements 
+        of the current one.
+
+        We require that the values of the attributes given by 
+        _satisfies_fields are the same.  A specobject with a value of None 
+        for one of these attributes is less specific than one with 
+        a specific value; the former cannot satisfy the latter, 
+        but the latter can satisfy the former.
+
+        TODO: Ensure we don't encounter the case where self is completely 
+        unspecified (all values are None).  Perhaps this is done by making 
+        sure that at least one of the _satisfies_fields cannot be None?
+        """
+        if not isinstance(other, self.__class__):
+            raise TypeError('incompatible specobject types')
+        for attr_name in self._satisfies_fields:
+            self_value = getattr(self, attr_name)
+            other_value = getattr(other, attr_name)
+            if self_value is None:
+                continue
+            if self_value != other_value:
+                return False
+        return True
```

Comment:
snippet accompanying removed code if you decide to roll back
```shell
$> git diff
diff --git a/niceman/distributions/base.py b/niceman/distributions/base.py
index 13f6f9d..7dbb07c 100644
--- a/niceman/distributions/base.py
+++ b/niceman/distributions/base.py
@@ -156,6 +156,11 @@ class SpecObject(object):
             if isinstance(self_v, SpecObject):
                 if not self_v.satisfied_by(other_v):
                     return differ("the other does not satisfy self")
+            elif isinstance(self_v, TypedList):
+                return all(
+                    any(self_v_e.satisfied_by(o) for o in other_v)
+                    for self_v_e in self_v
+                )
             else:
                 # For now (and otherwise) - just a simple identity check
                 if self_v != other_v:
```